blog/content/post/2024-07-25-odroid-h4-getting-started.md

```diff
@@ -0,0 +1,423 @@
+schematics is a rarity, almost non-existent, but this board actually [have
+one](https://wiki.odroid.com/_media/odroid-h4/hardware/adln-h4_sch_2024-0306.pdf)!
+Yes! Imagine what a firmware developer thinks seeing this... Of course,
+nothing else than open-source firmware port, obviously. In this blog post I
+would like to show you the perspective of firmware porting from scratch,
+beginning with the starting steps and the necessary preparation work you
+should do each time when dealing with new hardware.
+
+## Unboxing
+
+This is probably the most pleasant moment (except the final result of having
+the working open-source firmware build) of the whole process. Let's unpack
+this little baby and see what we got...
+
+The platform came to me with small 3 boxes.
+
+![Unboxing](/img/odroid/odroid_boxes.jpg)
+
+The white one has a power supply. It also has a switchable plug, it is useful
+when you travel and may need to switch the plug to other countries' standards.
+
+![Unboxing](/img/odroid/odroid_psu.jpg)
+
+In the smaller dark box there is a small bag containing an eMMC module with
+Hardkernel signature. Typically you see eMMCs embedded on the board itself.
+However, the size constraints may have forced the designers to put it on a
+separate module, or to simply lower the board's price. Cool, we will figure
+out how to connect it later.
+
+![Unboxing](/img/odroid/odroid_emmc.jpg)
+
+And now the shining start of our show. The board... It came packed in a
+electrostatic-proof bag.
+
+![Unboxing](/img/odroid/odroid_board1.jpg)
+
+Underneath it we have two coin cell batteries. As the Hardkernel page says:
+`BIOS Backup Battery (All H series models include a backup battery by default)`
+and they do include it indeed.
+
+![Unboxing](/img/odroid/odroid_batteries.jpg)
+
+The board itself is so small that it fits into my hand.
+
+![Unboxing](/img/odroid/odroid_board2.jpg)
+
+And mine already has a RAM stick plugged in. According to board
+specifications, it has a single channel DDR5 memory. In single channel
+configuration it may reach a pretty frequency of up to 4800MHz (depending on
+the RAM stick as well of course).
+
+![Unboxing](/img/odroid/odroid_board3.jpg)
+
+## First launch
+
+Before we attempt any work, let's see if it even works... Things may happen in
+during the shipping, so ensure it boots properly and there are no signs of
+damage, before you start being rough with it. And believe me, we will (just a
+bit). Let's plug one of the coin cell batteries so that our BIOS setting will
+not get reset every boot (typically a CMOS battery reset will cause booting
+with default settings, like in last century PCs). Also connect the eMMC module
+to check if it is detected properly by the BIOS and OS.
+
+![Plugging eMMC](/img/odroid/odroid_emmc_plug.jpg)
+
+First boot may take a couple of minutes and include board resets, so be
+patient. The board reste may be observed by the blue LEDs going off. When they
```

Comment:
reste?

Reply:
https://github.com/3mdeb/news-and-ideas/pull/599/commits/6a76815d92d3056aa5a55ff8fe3c039d8ec526b8#diff-a448115412de00a5d6d849f9b70b6f4a3bd46159e359ff67697ff44f3be6d186R100